src/modules/Dimmer/Dimmer.js

```diff
@@ -0,0 +1,187 @@
+import _ from 'lodash'
+import cx from 'classnames'
+import React, { Component, PropTypes } from 'react'
+
+import {
+  customPropTypes,
+  getElementType,
+  getUnhandledProps,
+  makeDebugger,
+  META,
+  useKeyOnly,
+} from '../../lib'
+import Portal from '../../addons/Portal'
+import DimmerDimmable from './DimmerDimmable'
+
+const debug = makeDebugger('dimmer')
+
+const _meta = {
+  name: 'Dimmer',
+  type: META.TYPES.MODULE,
+}
+
+/**
+ * A dimmer hides distractions to focus attention on particular content.
+ */
+class Dimmer extends Component {
+  static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** An active dimmer will dim its parent container. */
+    active: customPropTypes.every([
+      PropTypes.bool,
+      customPropTypes.disallow(['disabled']),
+    ]),
+
+    /** A dimmable element can blur its contents. */
+    blurring: customPropTypes.every([
+      PropTypes.bool,
+      customPropTypes.demand(['dimmable']),
+    ]),
+
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    closeOnOutsideClick: PropTypes.bool,
+
+    dimmable: customPropTypes.as,
+
+    dimmed: customPropTypes.every([
+      PropTypes.bool,
+      customPropTypes.demand(['dimmable']),
+    ]),
+
+    /** A disabled dimmer cannot be activated. */
+    disabled: customPropTypes.every([
+      PropTypes.bool,
+      customPropTypes.disallow(['active']),
+    ]),
+
+    /** A dimmer can be formatted to have its colors inverted. */
+    inverted: PropTypes.bool,
+
+    /** Called when a close event happens */
+    onClose: PropTypes.func,
+
+    page: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    closeOnOutsideClick: true,
+  }
+
+  static _meta = _meta
+
+  handlePageMount = () => {
+    debug('handleMount()')
+
+    document.body.classList.add('dimmed', 'dimmable')
```

Comment:
Not exactly, it added on `Portal`s mount, while `Portal` will mount on `active`. This was plan.